Mathlib/Data/ENat/Basic.lean

```diff
@@ -89,6 +89,16 @@ theorem coe_sub (m n : ℕ) : ↑(m - n) = (m - n : ℕ∞) :=
 @[simp] theorem mul_top (hm : m ≠ 0) : m * ⊤ = ⊤ := WithTop.mul_top hm
 @[simp] theorem top_mul (hm : m ≠ 0) : ⊤ * m = ⊤ := WithTop.top_mul hm
 
+theorem top_pow {n : ℕ} (n_pos : 0 < n) : (⊤ : ℕ∞)^n = ⊤ := by
+  apply @Nat.le_induction 1 (fun m : ℕ ↦ fun _ : 1 ≤ m ↦ (⊤ : ℕ∞) ^ m = ⊤) (pow_one ⊤)
+  · intro m _ h
+    calc
+      (⊤ : ℕ∞)^(m + 1) = ⊤^m * ⊤^1 := by rw [pow_add ⊤ m 1]
+                     _ = ⊤ * ⊤^1   := by rw [h]
+                     _ = ⊤ * ⊤     := by rw [pow_one ⊤]
+                     _ = ⊤         := WithTop.top_mul_top
+  · exact n_pos
```

Comment:
When the steps of the computation are trivial, it's often better to just use `simp`, like
```suggestion
  apply Nat.le_induction (P := fun m : ℕ ↦ fun _ : 1 ≤ m ↦ (⊤ : ℕ∞) ^ m = ⊤) (pow_one ⊤) _ _ n_pos
  intro m _ h
  simp only [pow_add, h, pow_one]
  exact WithTop.top_mul_top
```